Description: The firmware upload script tried to upload the firmware via
 the usblp kernel module and with the CUPS USB backend (using libusb) in
 parallel, making the printer crash.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1003843
Author: Till Kamppeter <till.kamppeter@gmail.com>
Last-Update: 2013-02-21

--- a/hplj1000
+++ b/hplj1000
@@ -261,8 +261,14 @@
     # force downloading to a specific device
     #
     load_usblp "$DEV"
-elif [ -x $PRINTERID -o -x $USB_BACKEND ]; then
-    if [ -x $PRINTERID ]; then
+elif [ -x $USB_BACKEND ]; then
+        #
+        # If we have CUPS installed, use the CUPS "usb" backend, as then we do
+        # not need to care whether the system uses the usblp kernel module or
+        # libusb 
+        #
+        load_cups
+elif [ -x $PRINTERID ]; then
 	#
 	# Sniff around for printers that need a firmware download
 	#
@@ -281,20 +287,6 @@
 		fi
 	    fi
 	done
-	if [ "$usblps" != " " ]; then
-	    # All done!
-	    exit
-	fi
-    fi
-
-    if [ -x $USB_BACKEND ]; then
-	#
-	# If we have CUPS installed, use the CUPS "usb" backend, as then we do
-	# not need to care whether the system uses the usblp kernel module or
-	# libusb
-	#
-	load_cups
-    fi
 else
     log "HP LaserJet $MODEL firmware was not downloaded..."
     log "...couldn't find $PRINTERID, DEV is not set, and CUPS not installed."
